### PR TITLE
SC-1740: atomic design updates

### PR DESCRIPTION
**Pull request by samcartzach**
- locks our ES Lint
- Moves the "display mode" to the first redux implementation
- Introduce some new "atoms"
- Introduce some new "molecules"
- Move some components into the correct directories
- Updates some basic tests
- Updates to basic SCSS/CSS

### CODE REVIEW

File: src/atoms/Image.jsx [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  }
+
+  render() {
+    return (
+      <img src={this.props.src} alt={this.props.alt} className="img-fluid" />
+    );
+  }
+}
+
+Image.propTypes = {
```
Comment (samcartalex) [inline src/atoms/Image.jsx]: If we're transpiling with babel, i think we can do

`static propTypes = {`

inside the class

File: src/atoms/Image.jsx [hunk 2]
```diff
@@ -0,0 +1,21 @@
+import React, { Component }  from 'react';
+import { string } from 'prop-types';
+
+export const Image = class Image extends Component {
```
Comment (samcartalex) [inline src/atoms/Image.jsx]: i think

`export class Image extends Component` 

is more readable if we're able to start using static for propTypes

File: src/organisms/Canvas.jsx [hunk 3]
```diff
@@ -28,44 +36,32 @@ export const Page = class Page extends Component {
...
-  handleCanvasViewChange(view) {
-    this.setState({
-      displayView: view
-    });
-  }
-
-  renderCanvasView() {
-    return this.state.displayView === 'mobile' ? "lb-mobileCanvas" : "";
+  renderCanvasView(currentView) {
+    return currentView === 'mobile' ? "lb-mobileCanvas" : "";
```
Comment (samcartalex) [inline src/organisms/Canvas.jsx]: nit: single ticks and 

`(currentView === 'mobile') ? `

for ternary expressions... definitely a personable preference, but whatever

File: src/organisms/Sidebar.jsx [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+
+  handleSave() {
+
+  };
+
+  handleTabClick() {
+    // store: dispatch set state of tab
+  }
+
+  _renderTabs() {
```
Comment (samcartalex) [inline src/organisms/Sidebar.jsx]: boo leading underscores

File: src/organisms/Sidebar.jsx [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+    const handleTabClick = this.handleTabClick;
+    // TODO: check if there are props
+    const activeTab = this.state.activeTab || 'tab-default';
+
+    // Add Tabs for individual content areas
+    return (
```
Comment (samcartalex) [inline src/organisms/Sidebar.jsx]: convert this in to a component, don't return jsx

File: src/reducers/changeCanvasViewReducer.js [hunk 6]
```diff
@@ -0,0 +1,10 @@
+export default (state = {}, action) => {
+  switch (action.type) {
+    case 'CHANGE_DISPLAY_VIEW':
+      return action.payload.displayView;
+    default:
+      return state;
+  }
+};
+
+export const getDisplayView = state => state.displayView;
```
Comment (samcartalex) [inline src/reducers/changeCanvasViewReducer.js]: this seems weird to me, mostly having something other than a reducer in the reducer file

when we did checkout redesign, we put a whole bunch of these little 1-3 liners in like a single file

File: src/organisms/Sidebar.jsx (hunk 4 above)
Comment (samcartzach) [reply src/organisms/Sidebar.jsx]: will update - was following a "private method" pattern which I don't strongly agree with

File: src/reducers/changeCanvasViewReducer.js (hunk 6 above)
Comment (samcartzach) [reply src/reducers/changeCanvasViewReducer.js]: per our conversation, leaving this here for now until we discuss and develop more reducers.

File: src/atoms/Widget.jsx [hunk 7]
```diff
@@ -0,0 +1,18 @@
+import React, { Component }  from 'react';
+import { string } from 'prop-types';
+
+const Widget = ({ name, image }) => (
```
Comment (samcartalex) [inline src/atoms/Widget.jsx]: just as an FYI, you can also use `PureComponent` for these... i think this way might be faster though

File: src/organisms/Canvas.jsx [hunk 8]
```diff
@@ -109,3 +103,15 @@ export const Page = class Page extends Component {
     );
   }
 };
+
+Canvas.propTypes = {
```
Comment (samcartalex) [inline src/organisms/Canvas.jsx]: you should move this to `static propTypes`

File: src/organisms/Sidebar.jsx [hunk 9]
```diff
@@ -0,0 +1,58 @@
+import React, { Component } from 'react';
+import {Tab} from '../atoms/Tab';
+import { SidebarHeader } from '../molecules/SidebarHeader';
+import {WidgetList} from '../molecules/WidgetList';
+import Widget from '../atoms/Widget';
+
+export const Sidebar = class Sidebar extends Component {
```
Comment (samcartalex) [inline src/organisms/Sidebar.jsx]: `export default class Sidebar`

File: src/organisms/Sidebar.jsx [hunk 10]
```diff
@@ -0,0 +1,58 @@
+import React, { Component } from 'react';
+import {Tab} from '../atoms/Tab';
```
Comment (samcartalex) [inline src/organisms/Sidebar.jsx]: lint?

File: src/store.js [hunk 11]
```diff
@@ -11,7 +11,7 @@ const finalCreateStore = compose(
   DevTools.instrument()
 )(createStore);
 
-module.exports = function configureStore(initialState) {
+export const configureStore = function configureStore(initialState) {
```
Comment (samcartalex) [inline src/store.js]: i think you can fat arrow this up

File: src/store.js [hunk 12]
```diff
@@ -22,3 +22,7 @@ module.exports = function configureStore(initialState) {
 
   return store;
 };
+
+export const initialState = {
+  displayView: 'desktop',
```
Comment (samcartalex) [inline src/store.js]: trailing comma

Comment (samcartzach) [reply src/store.js]: new standards like trailing commas :/

File: src/organisms/Sidebar.jsx (hunk 10 above)
Comment (samcartzach) [reply src/organisms/Sidebar.jsx]: it passed linting, so I don't think I have that rule in place. Updated this though since I prefer spaces.

**Pull request merged**